src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -634,12 +634,19 @@ def display_region_recommendation(cmd, namespace):
     telemetry.set_region_identified(namespace.location, identified_region)
 
     if identified_region and cmd.cli_ctx.config.getboolean('core', 'display_region_identified', True):
-        logger.warning('Selecting "%s" may reduce your costs. '
-                       'The region you\'ve selected may cost more for the same services. '
-                       'You can disable this message in the future with the command'
-                       ' "az config set core.display_region_identified=false". '
-                       'Learn more at https://go.microsoft.com/fwlink/?linkid=222571 ',
-                       identified_region)
+        from azure.cli.core.style import Style, print_styled_text
+        import sys
+        recommend_region = 'Selecting "' + identified_region + '" may reduce your costs. ' \
+                           'The region you\'ve selected may cost more for the same services. ' \
+                           'You can disable this message in the future with the command '
+        disable_config = '"az config set core.display_region_identified=false"'
+        learn_more_msg = '. Learn more at https://go.microsoft.com/fwlink/?linkid=222571 '
```

Comment:
I think it is better to split the message into parts like this:
```suggestion
        disable_config = '"az config set core.display_region_identified=false". '
        learn_more_msg = 'Learn more at https://go.microsoft.com/fwlink/?linkid=222571 '
```